cli: TestTenantZip failed

cli.TestTenantZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20282885?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20282885?buildTab=artifacts#/) on release-25.3 @ [8cd431de0178bc06eba0dc81c767eb4189761836](https://github.com/cockroachdb/cockroach/commits/8cd431de0178bc06eba0dc81c767eb4189761836):


```
=== RUN   TestTenantZip
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/ffde1473fe6ddd7d697975426d6a2b83/logTestTenantZip1168813759
    test_log_scope.go:76: use -show-logs to present logs inline
    zip_tenant_test.go:112: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/ffde1473fe6ddd7d697975426d6a2b83/logTestTenantZip1168813759
--- FAIL: TestTenantZip (297.51s)
=== RUN   TestTenantZip/testzip_external_process_virtualization
[debug --host=127.0.0.1:39873 --insecure=true zip --concurrency=1 --cpu-profile-duration=1s --validate-zip-file=false /dev/null]
    zip_tenant_test.go:104: 
          /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/6239/execroot/com_github_cockroachdb_cockroach/bazel-out/s390x-fastbuild/bin/pkg/cli/cli_test_/cli_test.runfiles/com_github_cockroachdb_cockroach/pkg/cli/testdata/zip/testzip_external_process_virtualization:1:
          zip
        output didn't match expected:
          @@ -13,10 +13,13 @@
           [cluster] retrieving SQL data for crdb_internal.kv_node_liveness: last request failed: ERROR: unimplemented: operation is unsupported within a virtual cluster (SQLSTATE 0A000)
           <dumping SQL tables>
           [cluster] retrieving SQL data for crdb_internal.kv_node_status: last request failed: ERROR: unimplemented: operation is unsupported within a virtual cluster (SQLSTATE 0A000)
           <dumping SQL tables>
           [cluster] retrieving SQL data for crdb_internal.kv_store_status: last request failed: ERROR: unimplemented: operation is unsupported within a virtual cluster (SQLSTATE 0A000)
          +<dumping SQL tables>
          +[cluster] retrieving SQL data for system.lease... writing output: debug/system.lease.txt...NOTICE: No license is installed. Throttling will begin after 2025-08-08 18:40:03 +0000 UTC unless a license is installed before then.
          + done
           <dumping SQL tables>
           [cluster] requesting nodes... received response... writing JSON output: debug/nodes.json... done
           [cluster] requesting liveness... received response... writing JSON output: debug/liveness.json... done
           [cluster] requesting tenant ranges... received response...
           [cluster] requesting tenant ranges: last request failed: rpc error: ...
--- FAIL: TestTenantZip/testzip_external_process_virtualization (93.31s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-53145

Epic CRDB-21133

--- a/pkg/cli/testutils.go
+++ b/pkg/cli/testutils.go
@@ -368,8 +368,9 @@ func captureOutput(f func()) (out string, err error) {
 		// both specific names and total number of system and internal tables that
 		// are exported. The regex matches the rest of the line after the prefix
 		// unless the line contains an uppercase E to avoid trimming "ERROR" message
-		// lines, which are expected (and tested) for certain tables.
-		out = regexp.MustCompile(`(.*retrieving SQL data for ([^E\n])*\n)+`).
+		// lines (but not NOTICE which is informational), which are expected
+		// (and tested) for certain tables.
+		out = regexp.MustCompile(`(.*retrieving SQL data for (([^E\n])*|.*NOTICE.*)\n)+`).
 			ReplaceAllString(out, "<dumping SQL tables>\n")
 	}()
 

